pkg/login/social/azuread_oauth.go

```diff
@@ -127,8 +148,64 @@ func hasRole(roles []string, role models.RoleType) bool {
 	return false
 }
 
-func extractGroups(claims azureClaims) []string {
-	groups := make([]string, 0)
-	groups = append(groups, claims.Groups...)
-	return groups
+type getAzureGroupRequest struct {
+	SecurityEnabledOnly bool `json:"securityEnabledOnly"`
+}
+
+type getAzureGroupResponse struct {
+	Value []string `json:"value"`
+}
+
+func extractGroups(client *http.Client, claims azureClaims, token *oauth2.Token) ([]string, error) {
+	if len(claims.Groups) > 0 {
+		return claims.Groups, nil
+	} else if claims.ClaimNames.Groups != "" {
```

Comment:
If you re-arrange this a little, you'll be able to shift the big block to the left 🤔

```suggestion
	}

	if claims.ClaimNames.Groups == "" {
		return nil, nil
	}
```

Reply:
true